pkg/delete/internal/document/delete.go

```diff
@@ -108,3 +107,19 @@ func isAPIErrorStatusNotFound(err error) bool {
 
 	return apiErr.StatusCode == http.StatusNotFound
 }
+
+func DeleteAll(ctx context.Context, c client) error {
+	listResponse, err := c.List(ctx, fmt.Sprintf("type='%s' or type='%s'", documents.Dashboard, documents.Notebook))
+	if err != nil {
+		return err
+	}
+
+	var retErr error
+	for _, x := range listResponse.Responses {
+		err := deleteSingle(ctx, c, pointer.DeletePointer{Type: x.Type, OriginObjectId: x.ID})
+		if err != nil {
+			retErr = errors.Join(retErr, err)
```

Comment:
I think it would be better to accumulate the errors first (for example in a slice) and then join them at the end outside of the loop

Reply:
In this case it doesn't have any difference.